cscs-checks/tools/profiling_and_debugging/intel_sde_berkeley_stream.py

```diff
@@ -0,0 +1,153 @@
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class SdeBaseTest(rfm.RegressionTest):
+    '''This test checks the values reported by Intel SDE for roofline modeling:
+       - https://software.intel.com/en-us/articles/
+            intel-software-development-emulator
+       - https://bitbucket.org/dwdoerf/stream-ai-example/src/master/
+       - https://www.nersc.gov/
+            users/application-performance/measuring-arithmetic-intensity
+    '''
+    def __init__(self):
+        super().__init__()
+        self.descr = 'Roofline Analysis test with Intel SDE'
+        self.sourcesdir = os.path.join(self.current_system.resourcesdir,
+                                       'roofline', 'sde')
+        self.build_system = 'SingleSource'
+        self.sourcepath = 'stream_mpi.c'
+        self.executable = 'sde'
+        self.target_executable = './stream.exe'
+        self.sde = '%s.sde' % self.target_executable
+        self.rpt = '%s.rpt' % self.target_executable
+        self.build_system.ldflags = ['-g', '-O3', '-qopenmp', '-restrict',
+                                     '-qopt-streaming-stores', 'always']
+        self.pre_run = [
+            'mv %s %s' % (self.executable, self.target_executable),
+            'module use /apps/dom/UES/jenkins/7.0.UP00/mc/easybuild/'
+            'experimental/modules/all',
+            'module load sde',
+            'sde -help'
+        ]
+        self.sanity_patterns = sn.assert_found('Total FLOPs =', self.rpt)
+        self.post_run = ['./parse-sde.sh %s.* &> %s' % (self.sde, self.rpt)]
+        self.maintainers = ['JG']
+        self.tags = {'scs'}
+
+    @property
+    @sn.sanity_function
+    def arithmetic_intensity(self):
+        flops = sn.extractsingle(r'^--->Total FLOPs = (?P<flops>\d+)',
+                                 self.rpt, 'flops', int)
+        byts = sn.extractsingle(r'^--->Total Bytes = (?P<byts>\d+)',
```

Comment:
There's no problem in naming this variable `bytes`, since you don't want the built-in `bytes` in this function.